### DISCUSSION TITLE
Make `lsp-format-buffer` respect editorconfig

### DISCUSSION THREAD

**Discussion by mangkoran** [emacs-lsp/lsp-mode#3308] [Q&A] [answered]
Hello everyone. I have an issue with editorconfig. Using `ts-ls` on `web-mode`, whenever I run `lsp-format-buffer`, it will format the buffer with 4 spaces. My editorconfig is shown [here](https://gist.github.com/mangkoran/23430bcb4f5523dea974cfe3ff479d16).
Currently, I set `lsp-enable-indentation` to `nil`. Whenever I want to format my buffer, I run `editorconfig-format-buffer` instead of `lsp-format-*`.

How to make `lsp-format-*` respect editorconfig?

**Comment by kiennq** [comment, marked as answer]
`lsp-mode` get the indentation width from `lsp--formatting-indent-alist`.
If you can find which indentation variable `web-mode` is using and `editorconfig` is modifying, you can add that to the alist and `lsp-mode` will ask the language server to format the buffer with the correct width.
Note that, it also depends on the language server to support indentation width or not.

**Reply by mangkoran** [reply]
I found the possibly variable. But then how do I add it to `lsp--formatting-indent-alist`? Sorry I am not to seasoned with Elisp yet. Is the below code correct to put in `init.el`?
```elisp
(after! lsp-mode
    (add-to-list 'lsp--formatting-indent-alist '(typescript-tsx-mode . typescript-indent-level)))
```
